File: src/test/console/ruby/AndroidAppTest.rb
require 'rubygems'
require 'test/unit'
require 'appium_lib'
require_relative './data/Device'
require_relative './config/config'

class AndroidAppTest < Test::Unit::TestCase
  def test_should_navigate_on_APIDemos_app
    device = getABookableDevice 'Android'

    options = {
      app: 'https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/ApiDemos-debug.apk',
      appPackage: 'io.appium.android.apis',
      appActivity: '.ApiDemos'
    }
    desired_caps = getAppCapabilitiesFor device, options
    @driver = Appium::Driver.new(desired_caps, false)

    begin
      @driver.start_driver
      puts "https://portal.kobiton.com/sessions/#{@driver.driver.capabilities['kobitonSessionId']}"
      @element = @driver.find_element(:class_name, 'android.widget.TextView')
      assert_equal('API DEMOS', @element.text.upcase)
    ensure
      @driver.driver.quit
    end
  end
end

File: javascript/android-app-test.js
import 'babel-polyfill'
import 'colors'
import wd from 'wd'
import {assert} from 'chai'

const username = 'your Kobiton username'
const apiKey = 'your Kobiton api key'

const kobitonServerConfig = {
  protocol: 'https',
  host: 'api.kobiton.com',
  auth: `${username}:${apiKey}`
}

const desiredCaps = {
  sessionName:        'Automation test session',
  sessionDescription: 'This is an example for Android app', 
  deviceOrientation:  'portrait',  
  captureScreenshots: true, 
  deviceGroup:        'KOBITON', 
  deviceName:         'Galaxy Note3',
  platformName:       'Android',
  app: 'https://appium.github.io/appium/assets/ApiDemos-debug.apk',
  appPackage: 'io.appium.android.apis',
  appActivity: '.ApiDemos'
}

let driver

describe('Android App sample', () => {

  before(async () => {
    driver = wd.promiseChainRemote(kobitonServerConfig)

    driver.on('status', (info) => {
      console.log(info.cyan)
    })
    driver.on('command', (meth, path, data) => {
      console.log(' > ' + meth.yellow, path.grey, data || '')
    })
    driver.on('http', (meth, path, data) => {
      console.log(' > ' + meth.magenta, path, (data || '').grey)
    })

    try {
      await driver.init(desiredCaps)
    }
    catch (err) {
      if (err.data) {
        console.error(`init driver: ${err.data}`)
      }
    throw err
    }
  })

  it('should show the app label', async () => {
    await driver.elementByClassName("android.widget.TextView")
      .text().then(function(text) {
        assert.equal(text, 'API Demos')
      })
  })

  after(async () => {
    if (driver != null) {
    try {
      await driver.quit()
    }
    catch (err) {
      console.error(`quit driver: ${err}`)
    }
  }
  })
})
